bot/cogs/filtering.py

```diff
@@ -368,6 +399,46 @@ def mod_log(self) -> ModLog:
         except discord.errors.Forbidden:
             await channel.send(f"{filtered_member.mention} {reason}")
 
+    async def _scheduled_task(self, msg: dict) -> None:
+        """A coroutine which delete the offensive message once the delete date is reached."""
+        delete_at = datetime.datetime.fromisoformat(msg['delete_date'][:-1])
+
+        await wait_until(delete_at)
+        await self.delete_offensive_msg(msg)
+
+        self.cancel_task(msg['id'])
+
+    async def reschedule_offensive_msg_deletion(self) -> None:
+        """Get all the pending message deletion from the API and reschedule them."""
+        await self.bot.wait_until_ready()
+        response = await self.bot.api_client.get(
+            'bot/offensive-message',
+        )
+
+        now = datetime.datetime.utcnow()
+        loop = asyncio.get_event_loop()
+
+        for msg in response:
+            delete_at = datetime.datetime.fromisoformat(msg['delete_date'][:-1])
+
+            if delete_at < now:
+                await self.delete_offensive_msg(msg)
+            else:
+                self.schedule_task(loop, msg['id'], msg)
+
+    async def delete_offensive_msg(self, msg: Mapping[str, str]) -> None:
+        """Delete an offensive message, and then delete it from the db."""
+        try:
+            channel = self.bot.get_channel(msg['channel_id'])
+            if channel:
+                msg_obj = await channel.fetch_message(msg['id'])
+                await msg_obj.delete()
+        except NotFound:
+            log.info(f"Tried to delete message {msg['id']}, but the message can't be found "
+                     f"(it has been probably already deleted).")
+        await self.bot.api_client.delete(f'bot/offensive-message/{msg["id"]}')
```

Comment:
A space here makes it easier to read

```suggestion

        await self.bot.api_client.delete(f'bot/offensive-message/{msg["id"]}')
```